cypress/e2e/homepage.spec.ts

```diff
@@ -0,0 +1,217 @@
+      } else if (viewport === "macbook-16") {
+        cy.findAllByTestId("header-getDfi-button").first().click();
+      }
+
+      cy.url().should("include", "/explore/dfi#get-dfi");
+    });
+
+    // No TC
+    it("should have expected Header title and description", () => {
+      cy.findByTestId("header-title").should(
+        "have.text",
+        "Connecting old and new worlds"
+      );
+
+      // No TC
+      cy.findByTestId("header-desc").should(
+        "have.text",
+        "A blockchain dedicated to fast, intelligent and transparent decentralized financial services, accessible by everyone."
+      );
+    });
+
+    // No TC
+    it("should have Start exploring section clickable", () => {
+      if (viewport === "iphone-xr") {
+        cy.findByTestId("statistic-title-TVL IN USD").should("be.visible");
+      } else if (viewport === "macbook-16") {
+        cy.findByTestId("startExploring-button").click({ force: true }); // covered by another element
+        cy.url().should("include", "/#statistics_display");
+        cy.findByTestId("statistic-title-TOTAL VALUE LOCKED IN USD").should(
+          "be.visible"
+        );
+      }
+
+      cy.findByTestId("statistic-title-TOTAL DFI MINTED").should("be.visible");
+      cy.findByTestId("statistic-title-MASTERNODES").should("be.visible");
+    });
+
+    // No TC
+    it("should have clickable Explore DeFi Meta Chain button", () => {
+      cy.findByTestId("explore-dmc-button-Explore DeFi Meta Chain")
+        .should("be.visible")
+        .and("have.attr", "href")
+        .and("include", "https://meta.defichain.com");
+    });
+
+    // No TC
+    it("should have clickable View all articles button", () => {
+      cy.findByTestId("explore-dmc-button-View all articles")
+        .should("be.visible")
+        .and("have.attr", "href")
+        .and("include", "https://blog.defichain.com");
+    });
+
+    // No TC
+    it("should have Ecosystem links displayed", () => {
+      ecosystemLinks.forEach((ecosystemItem) => {
+        cy.findByTestId(
+          `ecosystem-link-${ecosystemItem.testid}`
+        ).scrollIntoView();
+        cy.findByTestId(`ecosystem-link-${ecosystemItem.testid}`)
+          .should("contain", ecosystemItem.title)
+          .and("be.visible")
+          .and("have.attr", "href")
+          .and("include", ecosystemItem.link);
+      });
+    });
+
+    // No TC
+    it("should have expected Blockchain feature section title and description", () => {
+      cy.findByTestId("section-title-NON-TURING-COMPLETE BLOCKCHAIN").should(
+        "be.visible"
+      );
+      cy.findByTestId(
+        "section-header-Designed,for,Decentralized,Finance"
+      ).should("be.visible");
+      cy.findByTestId("section-desc-Designed,for,Decentralized,Finance")
+        .should("be.visible")
+        .and(
+          "have.text",
+          "DeFiChain offers comprehensive functionality tailored to the Distributed Ledger Technology (DLT) " +
```

Comment:
Can we make this one line?

Reply:
if 173 characters in one line is okay then can